### PR TITLE
Subsriptions: Adding base score

### PR DESCRIPTION
**Pull request by prateekbh**
- add baseScore capability 
- add score weight from config

### CODE REVIEW

File: extensions/amp-subscriptions-google/0.1/amp-subscriptions-google.js [hunk 1]
```diff
@@ -220,6 +223,10 @@ export class GoogleSubscriptionsPlatform {
       });
     }
   }
+
+  getBaseScore() {
```
Comment (dvoytenko) [inline extensions/amp-subscriptions-google/0.1/amp-subscriptions-google.js]: `/** @override */`

File: extensions/amp-subscriptions/0.1/local-subscription-platform.js [hunk 2]
```diff
@@ -226,6 +226,10 @@ export class LocalSubscriptionPlatform {
   supportsCurrentViewer() {
     return false;
   }
+
+  getBaseScore() {
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/local-subscription-platform.js]: Seems duplicated. Perhaps still worth it to just read it in `PlatformStore`?

File: extensions/amp-subscriptions/0.1/amp-subscriptions.js [hunk 3]
```diff
@@ -304,7 +305,8 @@ export class SubscriptionService {
         this.pageConfig_.getProductId(),
         'Product id is null'
     ));
-    this.platformStore_ = new PlatformStore(serviceIds);
+    this.platformStore_ = new PlatformStore(serviceIds,
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/amp-subscriptions.js]: Few lines here are clear duplicates of non-delegated case. See if you can factor it out.

File: extensions/amp-subscriptions/0.1/platform-store.js [hunk 4]
```diff
@@ -273,14 +278,17 @@ export class PlatformStore {
...
+        weight += 100000;
       }
 
+      // Add the base score
+      weight += platform.getBaseScore();
+
       // If supports the current viewer, gains weight 9
       if (preferViewerSupport && platform.supportsCurrentViewer()) {
-        weight += 9;
+        weight += this.scoreConfig_.supportsViewer;
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/platform-store.js]: Should be in `quotes` if you continue with `scoreConfig` as a `!JsonObject`.

File: extensions/amp-subscriptions/amp-subscriptions.md [hunk 5]
```diff
@@ -151,6 +154,13 @@ Based on `preferViewerSupport` (default: true) this document will give extra pre
 If you'd like to test the document's behavior in the context of a particular viewer, you can add `#viewerUrl=` fragment parameter. For instance, `#viewerUrl=https://www.google.com` would emulate a document's behavior inside a Google viewer.
 
 
+## Selecting platform
+So if no platforms are selected, we compete all the platforms based on platforms like
+
+1. Does the platform support the Viewer
+
+You can add `baseScore` < 100 key in any service configuration in case you want to increase baseScore of any platform so that it wins over other score evaluation factors.
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/amp-subscriptions.md]: Apply code quotes to all `baseScore` instances.

File: extensions/amp-subscriptions/amp-subscriptions.md [hunk 6]
```diff
@@ -139,6 +139,9 @@ The `amp-subscriptions` extension must be configured using JSON configuration:
       // Service 2 (a vendor service)
     }
   ],
+  "score": {
+    "supportsViewer": 10
+  }
   "preferViewerSupport": true
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/amp-subscriptions.md]: Should we deprecate/remove `preverViewerSupport` now?

File: extensions/amp-subscriptions/amp-subscriptions.md [hunk 7]
```diff
@@ -139,6 +139,9 @@ The `amp-subscriptions` extension must be configured using JSON configuration:
       // Service 2 (a vendor service)
     }
   ],
+  "score": {
+    "supportsViewer": 10
+  }
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/amp-subscriptions.md]: Comma

File: extensions/amp-subscriptions/0.1/amp-subscriptions.js (hunk 3 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/amp-subscriptions.js]: This is no more the same case after last PR

File: extensions/amp-subscriptions/0.1/platform-store.js [hunk 8]
```diff
@@ -53,6 +53,11 @@ export class PlatformStore {
 
     /** @private {!Array<string>} */
     this.failedPlatforms_ = [];
+
+    /** @private @const {!Object<string, number>} */
+    this.scoreConfig_ = Object.assign({
+      'supportsViewer': 9,
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/platform-store.js]: Now that it's public, let's make this a round value - `10`. It's documented as `10` anyway :)

**Pull request merged**